.github/workflows/deploy.yml

```diff
@@ -0,0 +1,43 @@
+name: Deploy to GitHub Pages
+
+on:
+  push:
+    branches:
+      - main
+      - 'docs/**'
+      - '.github/workflows/deploy.yml'
```

Comment:
I think we only need to list the develop branch here.

Reply:
But Guy told me that the update should only trigger when the PR is merged to the main.

Reply:
I see, okay. But what about the other two items? do we need them as target branches?

Reply:
no those are the target subdirectories and files, to which only if update is made then only the update takes place.

Reply:
I think you should be able to remove the other two and only keep main as the target branch for this workflow.

Reply:
Right, for that we would need to use the `paths` event under `push`. Something like:

```
on:
    push:
        paths:
          - 'docs/**'
```

Under `branches` it should only be main

Reply:
See: https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#using-filters-to-target-specific-paths-for-pull-request-or-push-events

Reply:
sorry my bad, wanted to do that, mistakely wrote that under the branches. corrected now